src/components/NavBar/NavigationBar.tsx

```diff
@@ -108,18 +109,39 @@ const SignOutButtonContainer = styled(NavElementContainer)`
 
 const LoginDependent: React.FC<Props> = (props) => {
     const pathname = usePathname();
-
     if (pathname === "/login") {
         return <></>;
     }
-
     return <>{props.children}</>;
 };
 
 interface Props {
     children?: React.ReactNode;
 }
 
+interface RoleProps {
+    children?: React.ReactNode;
+    role: string;
+    pathname: string;
+}
+
+const RoleDependent: React.FC<RoleProps> = ({ children, role, pathname }) => {
+    const hiddenPages = role
+        ? rolesAndHiddenPages[role]
+        : ["/clients", "/lists", "/calendar", "/admin"];
```

Comment:
just curious: is this list something we have to change everytime we add a new page to the navbar?

Reply:
Yes! So this list is for when the user is logged out, which ones to not show

Reply:
Isn't all pages except login then?

Reply:
Yes, but we only need the ones in the navbar here